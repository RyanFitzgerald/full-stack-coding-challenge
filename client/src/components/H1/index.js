import styled from 'styled-components';

const H1 = styled.h1`
  color: #373958;
  font-size: 2.5em;
  line-height: 1.25;
  margin: 0;
`;

export default H1;