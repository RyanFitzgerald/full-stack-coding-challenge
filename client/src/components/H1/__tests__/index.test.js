import React from 'react';

import H1 from '../index';

describe('<H1 />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <H1>Test</H1>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct element', () => {
    const renderedComponent = shallow(
      <H1/>
    );
    
    expect(renderedComponent.type()).toEqual('h1');
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <H1>{children}</H1>
    );
    
    expect(renderedComponent.contains(children)).toBe(true);
  });
});