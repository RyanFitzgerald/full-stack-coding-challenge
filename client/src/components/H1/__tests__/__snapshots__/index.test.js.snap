// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> should match the snapshot 1`] = `
.c0 {
  color: #373958;
  font-size: 2.5em;
  line-height: 1.25;
  margin: 0;
}

<h1
  className="c0"
>
  Test
</h1>
`;
